hil/api.py

```diff
@@ -130,9 +136,27 @@ def project_detach_node(project, node):
     for nic in node.nics:
         if nic.current_action is not None:
             raise errors.BlockedError("Node has pending network actions")
+    if (cfg.has_option('maintenance', 'enable') and
+            cfg.get('maintenance', 'enable')):
+        if (not cfg.has_option('maintenance', 'maintenance_project')):
+            raise errors.NotFoundError("Maintenance project not in hil.cfg.")
+        maintenance_proj = _must_find(
+                model.Project,
+                cfg.get('maintenance', 'maintenance_project')
+                )
+        if (project != maintenance_proj):
+            maintaining = True
+            if (not cfg.has_option('maintenance', 'url')):
+                raise errors.NotFoundError("Maintenance URL not in hil.cfg.")
+            else:
+                url = cfg.get('maintenance', 'url')
+            payload = json.dumps({'node': node_json})
     node.obm.stop_console()
     node.obm.delete_console()
     project.nodes.remove(node)
+    if (maintaining):
+        maintenance_proj.nodes.append(node)
+        requests.post(url, data=payload)
```

Comment:
why not move this block in the else part of the block where you get the url (line 151)? that way you wouldn't need tthe `maintaining` variable.

Reply:
Not sure if worth it, but should we get the response from the POST you do and then check if was succesful or not? (response >= 400?); and then either raise an error or write it to logs.

Reply:
Yeah, checking the response is a good idea.  Also the reason I added the block there is because we need to catch all errors before removing the node from its project, and only then can we move it to the maintenance pool.

Reply:
There's enough logic here that it might make sense to split it off into a helper function. That said, I'd do the change I suggested in the config file first, since it's going to affect the logic a bit.